oak-run-elastic/src/main/java/org/apache/jackrabbit/oak/index/ElasticIndexCommand.java

```diff
@@ -216,33 +207,25 @@ private void reindex(ElasticIndexOptions indexOpts, IndexHelper indexHelper, Str
                     indexOpts.getElasticScheme(), indexOpts.getElasticHost(),
                     indexOpts.getElasticPort(), indexOpts.getApiKeyId(), indexOpts.getApiKeySecret())) {
                 indexer.reindex();
-                // Wait for default flush interval before exiting the try block
-                // to make sure the client is not closed before the last flush
-                // TODO : See if this can be handled in a better manner
-                Thread.sleep(ElasticIndexDefinition.BULK_FLUSH_INTERVAL_MS_DEFAULT * 2);
-            } catch (InterruptedException e) {
-                log.debug("Exception while waiting for Elastic connection to close", e);
             }
         } else {
             try (ElasticOutOfBandIndexer indexer = new ElasticOutOfBandIndexer(indexHelper, indexerSupport, indexOpts.getIndexPrefix(),
                     indexOpts.getElasticScheme(), indexOpts.getElasticHost(),
                     indexOpts.getElasticPort(), indexOpts.getApiKeyId(), indexOpts.getApiKeySecret())) {
 
                 indexer.reindex();
-                // Wait for default flush interval before exiting the try block
-                // to make sure the client is not closed before the last flush
-                Thread.sleep(ElasticIndexDefinition.BULK_FLUSH_INTERVAL_MS_DEFAULT * 2);
-            } catch (InterruptedException e) {
-                log.debug("Exception while waiting for Elastic connection to close", e);
             }
         }
         indexerSupport.writeMetaInfo(checkpoint);
+        File destDir = indexerSupport.copyIndexFilesToOutput();
```

Comment:
This is copying the metadata file, right? (Not sure if I understand the logic)

Reply:
So basically the metadata files are initially created in the working repo (basically where the repo's temp files etc are created for the oak nodestore created). We copy the metadata files from this directory to a specified output directory because the working repo is cleaned up before starting any oak-run command (to clear up earlier cache etc), so if we don't copy , these files would also be cleared. 

This is the same with lucene,  initially this was not really needed for elastic, since we never used these meta file - but now since we are using the Import command to apply elastic index def and do the catchup - we need these files. So had to add this copy call here too.

Reply:
Added more description in the code comment as well.